lib/segment/src/common/rocksdb_buffered_delete_wrapper.rs

```diff
@@ -60,4 +81,34 @@ impl DatabaseColumnScheduledDeleteWrapper {
     pub fn lock_db(&self) -> LockedDatabaseColumnWrapper {
```

Comment:
It's not related to this PR, but maybe needs an issue. I think we have to remove `lock_db` and because user get access to rocksdb data which may be obsolete because of non-applied changes

Reply:
We currently use it in a few places, notably for `iter()`. It does not look like we can easily remove it because of lifetime issues with these iterators.